pkg/infrastructure/aws/clusterapi/aws.go

```diff
@@ -24,3 +34,112 @@ func (*Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionInp
 	}
 	return nil
 }
+
+// InfraReady creates private hosted zone and DNS records.
+func (*Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput) error {
+	awsCluster := &capa.AWSCluster{}
+	key := k8sClient.ObjectKey{
+		Name:      in.InfraID,
+		Namespace: capiutils.Namespace,
+	}
+	if err := in.Client.Get(ctx, key, awsCluster); err != nil {
+		return fmt.Errorf("failed to get AWSCluster: %w", err)
+	}
+
+	awsSession, err := in.InstallConfig.AWS.Session(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get aws session: %w", err)
+	}
+
+	subnetIDs := make([]string, 0, len(awsCluster.Spec.NetworkSpec.Subnets))
+	for _, s := range awsCluster.Spec.NetworkSpec.Subnets {
+		if s.IsPublic {
+			subnetIDs = append(subnetIDs, s.ResourceID)
+		}
+	}
+
+	vpcID, err := getVPCFromSubnets(ctx, awsSession, awsCluster.Spec.Region, subnetIDs)
```

Comment:
Considering all subnets belongs to the same VPC, what about to get a single subnet and query it's VPC ID?

Or maybe getting the VPC ID from infra? Is it available in `awsCluster` in this part? From temp control plane:
```
]$ oc get awscluster -o json -A | jq '.items[].spec.network.vpc' 
{
  "availabilityZoneSelection": "Ordered",
  "availabilityZoneUsageLimit": 3,
  "cidrBlock": "10.0.0.0/16",
  "id": "vpc-03e9b46a99bf53759",
  "internetGatewayId": "igw-067e33d25c35ffa88",
  "tags": {
    "Name": "mrb-capa-52-5n92s-vpc",
    "kubernetes.io/cluster/mrb-capa-52-5n92s": "owned",
    "sigs.k8s.io/cluster-api-provider-aws/cluster/mrb-capa-52-5n92s": "owned",
    "sigs.k8s.io/cluster-api-provider-aws/role": "common"
  }
}

```

Unfortunately, I can't see anything related to VPC ID in `.Status`. Thoughts: should we need to ask CAPA to add it? + @patrickdillon

Reply:
@r4f4 I can see you getting from spec[1] in the latest version. For me this thread is resolved, no more questions from me.
https://github.com/openshift/installer/blob/c423d63d3b3919538333cfb675070a131f474d90/pkg/infrastructure/aws/clusterapi/aws.go#L59